### PR TITLE
PR: Make middle mouse button click on editor tabs close the intended tab

### LINKED ISSUE #12558: Middle mouse button click on editor tabs may close wrong tab
Found a minor UI issue: When clicking the middle mouse button on the filename in an editor tab, it closes that tab, as expected. However, when doing this anywhere on the "X" symbol part of the tab, Spyder closes the next tab to the right.

E.g. in this image: 
Untitled
Middle mouse button click in the red area should close "temp.py", but it closes "untitled1.py". 
(Literal) Edge case: Clicking in the green area should close "untitled1.py", but does nothing.

Python 3.7.7 64-bit | Qt 5.9.6 | PyQt5 5.9.2 | Windows 10
clean miniconda environment & spyder --reset

### PR DESCRIPTION
**Pull request by steff456**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Add offset to remain the region of tab given an event and be able to close intended tab

tab

### Issue(s) Resolved

Fixes #12558 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Steff456

### CODE REVIEW

**Comment by ccordoba12** [comment]
Nice job @steff456!

@dalthviz, please test this PR on Windows to see if it's working as expected there. I'll test this one on my KDE desktop too.

**Comment by dalthviz** [comment]
@ccordoba12 it works for me on Windows :+1:

File: spyder/widgets/tabs.py [hunk 1]
```diff
@@ -352,16 +352,34 @@ def add_corner_widgets(self, widgets, corner=Qt.TopRightCorner):
...
+        Add offset to position event to capture the mouse cursor
+        inside a tab.
+        """
+        # This is necessary because self.tabBar().tabAt(event.pos()) is not
+        # returning the expected index. For further information see
+        # spyder-ide/spyder#12617
+        point = event.pos()
+        if sys.platform == 'darwin':
+            # The close button on tab is on the left
+            point.setX(point.x() + 3)
```
Comment (goanpeca) [inline spyder/widgets/tabs.py:366]: @steff456 and @dalthviz could  you check if this works as well if we are using the high DPI scaling property to make things bigger/smaller?

Comment (steff456) [reply spyder/widgets/tabs.py:366]: It is working for me in fullscreen mode

Comment (ccordoba12) [reply spyder/widgets/tabs.py:366]: @steff456, Gonzalo means if this is still working when you go to

    Preferences > General > Screen resolution

and set a custom resolution of 1.5 or 2 there.

Comment (dalthviz) [reply spyder/widgets/tabs.py:366]: Works for me on Windows :+1: (although maybe should I try to test this on Mac and/or changing the OS scale instead?)

**Comment by ccordoba12** [comment]
@steff456, please don't forget to check if this works with a change in the scaling factor of your screen so we can merge it.

File: spyder/widgets/tabs.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/widgets/tabs.py:366]: Thanks @dalthviz for your help!

**Comment by ccordoba12** [comment]
> @steff456, please don't forget to check if this works with a change in the scaling factor of your screen so we can merge it.

@steff456 confirmed with me that this PR works for her on macOS with different scale factors, so I'd say this is ready.

**Pull request merged**